typings/index.d.ts

```diff
@@ -171,8 +171,8 @@ declare namespace commander {
      *
      * @returns `this` command for chaining
      */
-    option(flags: string, description?: string, defaultValue?: string | boolean): this;
-    option(flags: string, description: string, regexp: RegExp, defaultValue?: string | boolean): this;
+    option(flags: string, description?: string, defaultValue?: any): this;
+    option(flags: string, description: string, regexp: RegExp, defaultValue?: any): this;
```

Comment:
The TypeScript is deliberately describing the most common cases. This signature is when you don't supply a coercion function, so are much more likely to be specifying a boolean or a string. (You can pass anything which is useful in rare cases, but if we describe that in the signature then we get no type checking!)

See #1119 for a lot of discussion, which led to the current signature.

I suggest the signature you want to be use in the TypeScript tests is the one with a coercion function.

Reply:
@shadowspawn I see what you mean regarding the RegExp coercion function, but is that really the case if you don't supply any coercion function.

The TypeScript compiler produces an error for me when providing an object instead of a string or boolean, there is just no runtime checking.

Reply:
@shadowspawn looks like I was confused, I see your point here.